sass/_vars.scss

```diff
@@ -1,113 +1,305 @@
 /* Copyright: Ankitects Pty Ltd and contributors
- * License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html */
+ * License: GNU AGPL, version 3 or later, http://www.gnu.org/licenses/agpl.html */
+
+@use "sass:map";
+@use "functions" as *;
+@use "colors" as *;
+
+@function palette($key, $shade) {
+    $color: map.get($colors, $key);
+    @return map.get($color, $shade);
+}
+
+$vars: (
+    props: (
+        border-radius: (
+            default: 5px,
+        ),
+    ),
+    colors: (
+        fg: (
+            default: (
+                light: palette(darkgray, 9),
+                dark: palette(lightgray, 0),
+            ),
+            subtle: (
+                light: palette(darkgray, 6),
+                dark: palette(lightgray, 3),
+            ),
+            disabled: (
+                light: palette(darkgray, 3),
+                dark: palette(lightgray, 6),
+            ),
+            faint: (
+                light: palette(darkgray, 1),
+                dark: palette(lightgray, 8),
+            ),
+        ),
+        canvas: (
+            default: (
+                light: palette(lightgray, 3),
+                dark: palette(darkgray, 5),
+            ),
+            elevated: (
+                light: white,
+                dark: palette(darkgray, 4),
+            ),
+            inset: (
+                light: palette(lightgray, 4),
+                dark: palette(darkgray, 6),
+            ),
+            overlay: (
+                light: white,
+                dark: black,
+            ),
+        ),
+        border: (
+            default: (
+                light: palette(lightgray, 7),
+                dark: palette(darkgray, 7),
+            ),
+            subtle: (
+                light: palette(lightgray, 6),
+                dark: palette(darkgray, 6),
+            ),
+            focus: (
+                light: palette(blue, 5),
+                dark: palette(blue, 5),
+            ),
+        ),
+        button: (
+            border: (
+                light: palette(lightgray, 6),
+                dark: palette(darkgray, 6),
+            ),
+            focus: (
+                default: palette(cyan, 3),
+                light: palette(cyan, 2),
+                dark: palette(cyan, 4),
```

Comment:
I'm a bit confused by default here - when is it used instead of light/dark?

Reply:
That's a leftover from experiments. I removed it :)